src/connector/src/sink/big_query.rs

```diff
@@ -308,83 +342,159 @@ impl BigQuerySinkWriter {
         pk_indices: Vec<usize>,
         is_append_only: bool,
     ) -> Result<Self> {
-        let client = config.common.build_client(&config.aws_auth_props).await?;
+        let client = config
+            .common
+            .build_writer_client(&config.aws_auth_props)
+            .await?;
+        let mut descriptor_proto = build_protobuf_schema(
+            schema
+                .fields()
+                .iter()
+                .map(|f| (f.name.as_str(), &f.data_type)),
+            config.common.table.clone(),
+        )?;
+
+        if !is_append_only {
+            let field = FieldDescriptorProto {
+                name: Some(CHANGE_TYPE.to_string()),
+                number: Some((schema.len() + 1) as i32),
+                r#type: Some(field_descriptor_proto::Type::String.into()),
+                ..Default::default()
+            };
+            descriptor_proto.field.push(field);
+        }
+
+        let descriptor_pool = build_protobuf_descriptor_pool(&descriptor_proto);
+        let message_descriptor = descriptor_pool
+            .get_message_by_name(&config.common.table)
+            .ok_or_else(|| {
+                SinkError::BigQuery(anyhow::anyhow!(
+                    "Can't find message proto {}",
+                    &config.common.table
+                ))
+            })?;
+        let proto_field = if !is_append_only {
+            let proto_field = message_descriptor
+                .get_field_by_name(CHANGE_TYPE)
```

Comment:
We can reuse the field we created in the code above instead of getting the field again by field name.